### PR TITLE
Add Kubernetes component

### PR DESCRIPTION
**Pull request by atanasdinov**
- Bootstraps Kubernetes component
- Introduces single node fully airgapped RKE2 setup
- Follow ups will include:
   - Revisiting kubernetes definition i.e. dropping fields such as CNI and multus which can be extracted from the config file
   - HA support
   - k3s support
   - Documentation 😬

### CODE REVIEW

File: pkg/combustion/templates/15-rke2-installer.sh.tpl
Comment (jdob) [inline pkg/combustion/templates/15-rke2-installer.sh.tpl:24]: What's the purpose for adding this to `PATH`? Is it just to get access to `kubectl` when you log into the box?

If this isn't needed for RKE2 to actually run in the first place, I think we should at least consider removing this. Generally speaking, production machines are locked down to all hell, so having this in `PATH` may be frowned upon by the more anal-retentive security admins.

Adding @rdoxenham for both his input based on his Linux experience and ultimately his recommendation as PM.

File: pkg/combustion/kubernetes.go
Comment (jdob) [inline pkg/combustion/kubernetes.go:136]: Is there a reason you manually used `errors.Is` instead of `os.IsNotExist`? I'm sure it's the same thing under the covers (too lazy to check), but feels safer to rely on the `os` package since that's where `Stat?()` came from in the first place.

File: pkg/combustion/kubernetes_test.go
Comment (jdob) [inline pkg/combustion/kubernetes_test.go:24]: I'm personally not a fan of any sort of panic/exit in the middle of unit tests; I think they should always run to completion. It's similar in my head to using `assert` instead of `require` where it makes sense. But it's not a big enough problem to warrant blocking the merge.

**Review by jdob — COMMENTED** [review]
Minor comments. The only one stopping me from approving is the conversation around if the PATH entry is an issue.

File: pkg/combustion/kubernetes.go
Comment (atanasdinov) [reply pkg/combustion/kubernetes.go:136]: If you take a look at the documentation of `os.IsNotExist`, it says:
```
// IsNotExist returns a boolean indicating whether the error is known to
// report that a file or directory does not exist. It is satisfied by
// ErrNotExist as well as some syscall errors.
//
// This function predates errors.Is. It only supports errors returned by
// the os package. New code should use errors.Is(err, fs.ErrNotExist).
```

This is also the reason why I might have changed such usages in the past.

File: pkg/combustion/kubernetes_test.go
Comment (atanasdinov) [reply pkg/combustion/kubernetes_test.go:24]: It is in fact a sanity check which prevents a misuse of the mock itself. We often rely on the returned error as a value to determine whether or not the mock operation is successful so distinguishing intentional errors with a panic is usually the way to go in these cases.

File: pkg/combustion/kubernetes.go
Comment (jdob) [reply pkg/combustion/kubernetes.go:136]: Good to know :)

File: pkg/combustion/templates/15-rke2-installer.sh.tpl
Comment (atanasdinov) [reply pkg/combustion/templates/15-rke2-installer.sh.tpl:24]: I indeed use it for the `kubectl` usages and I guess we can live without it. Makes the development stage just slightly easier though. I'm fine with dropping it.

Comment (agracey) [reply pkg/combustion/templates/15-rke2-installer.sh.tpl:24]: I would suggest removing it from the code but adding the steps in a "setups for development" best practices doc

Comment (rdoxenham) [reply pkg/combustion/templates/15-rke2-installer.sh.tpl:24]: Agreed, we should drop this from EIB. Interestingly, the k3s installer does this automagically, but rke2 doesn't. Either way, this should be a post-deployment action as/when/if required, not something we do by default.

Comment (agracey) [reply pkg/combustion/templates/15-rke2-installer.sh.tpl:24]: I've asked about it before and was told K3s does this by default because it's more likely used in local dev environment.

**Review by rdoxenham — APPROVED** [review]
LGTM and tested. Thanks!

**Review by jdob — APPROVED** [review]
Glad we came to a quick decision on the PATH thing. Approved.

**Pull request merged**
